actions/lib/formatters.py

```diff
@@ -116,36 +120,93 @@ def pull_to_dict(pull):
 
 
 def commit_to_dict(commit):
-    result = {}
-    result['sha'] = commit.sha
+    result = {'sha': commit.sha}
     return result
 
 
 def label_to_dict(label):
-    result = {}
+    result = {'name': label.name, 'color': label.color, 'url': label.url}
 
-    result['name'] = label.name
-    result['color'] = label.color
-    result['url'] = label.url
     return result
 
 
 def user_to_dict(user):
     if not user:
         return None
 
-    result = {}
-    result['name'] = user.name
-    result['login'] = user.login
+    result = {'name': user.name, 'login': user.login}
     return result
 
 
 def team_to_dict(team):
     if not team:
         return None
 
-    result = {}
-    result['id'] = team.id
-    result['name'] = team.name
-    result['members_count'] = team.members_count
+    result = {'id': team.id, 'name': team.name, 'members_count': team.members_count}
+    return result
+
+
+def contents_to_dict(contents, decode=False):
+    if not contents:
+        return None
+
+    directory = False
+    if isinstance(contents, list):
+        directory = True
+    else:
+        contents = [contents]
+
+    result = []
+    data = {}
+
+    for item in contents:
+        item_type = item.type
+        data['type'] = item_type
+        if item_type == 'symlink':
+            data['target'] = item.target
+        elif item_type == 'submodule':
+            data['submodule_git_url'] = item.submodule_git_url
+        elif not directory:
+            encoding = item.encoding
+            content = item.content
+            data['encoding'] = encoding
+            if decode and encoding == 'base64':
+                content = decode_base64(content)
+            data['content'] = content
+
+        data['size'] = item.size
+        data['name'] = item.name
+        data['path'] = item.path
+        data['sha'] = item.sha
+        data['url'] = item.url
+        data['git_url'] = item.git_url
+        data['html_url'] = item.html_url
+        data['download_url'] = item.download_url
+        result.append(data)
+
+    if not directory:
+        return result[0]
+    else:
+        return result
+
+
+def decode_base64(data):
+    """Decode base64, padding being optional.
+
+    :param data: Base64 data as an ASCII byte string
+    :returns: The decoded byte string.
+
+    """
+    missing_padding = len(data) % 4
```

Comment:
this seems a bit wonky, a comment here would be nice

Reply:
I dunno if it's still the case, but it stems from times where decoding with incorrect padding would give you an exception, see https://gist.github.com/perrygeo/ee7c65bb1541ff6ac770 or https://stackoverflow.com/questions/2941995/python-ignore-incorrect-padding-error-when-base64-decoding

Just to be safe.